kaolin/datasets/modelnet.py

```diff
@@ -12,63 +12,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Callable, Iterable, Optional, Union, List
+from typing import Iterable, Optional
 
 import torch
 import os
 from glob import glob
-from tqdm import tqdm
 
 from kaolin.rep.TriangleMesh import TriangleMesh
-from kaolin.transforms import transforms as tfs
 
 
 class ModelNet(object):
-    r""" Dataset class for the ModelNet dataset.
+    r"""Dataset class for the ModelNet dataset.
 
     Args:
-        basedir (str): Path to the base directory of the ModelNet dataset.
+        root (str): Path to the base directory of the ModelNet dataset.
         split (str, optional): Split to load ('train' vs 'test',
             default: 'train').
         categories (iterable, optional): List of categories to load
             (default: ['chair']).
-        transform (callable, optional): A function/transform to apply on each
-            loaded example.
-        device (str or torch.device, optional): Device to use (cpu,
-            cuda, cuda:1, etc.).  Default: 'cpu'
 
     Examples:
-        >>> dataset = ModelNet(basedir='data/ModelNet')
+        >>> dataset = ModelNet(root='data/ModelNet')
         >>> train_loader = DataLoader(dataset, batch_size=10, shuffle=True, num_workers=8)
         >>> obj, label = next(iter(train_loader))
     """
 
-    def __init__(self, basedir: str,
-                 split: Optional[str] = 'train',
-                 categories: Optional[Iterable] = ['bed'],
-                 transform: Optional[Callable] = None,
-                 device: Optional[Union[torch.device, str]] = 'cpu'):
+    def initialize(self, root: str,
+                   split: Optional[str] = 'train',
+                   categories: Optional[Iterable] = ['bed']):
```

Comment:
There shouldn't be any default value for categories.

On a side note, if you want to set a default argument value it shouldn't be a mutable value, it should be `None` and then the default value is set inside the function.

Reply:
Done.

Reply:
An update, can you set all categories by default ?